File: Practice/24/C++/codejson.cpp
#include <iostream>
#include <fstream>
#include "json.hpp"
#include <iomanip>
using namespace std;
using json = nlohmann::json;
int main(){
    std::ifstream input("in.json");
    std::ofstream output("out.json");
    json jin; input >> jin;
    json jout;
    for (auto& ochered : jin) 
    {
    int userID = ochered["userId"];
    bool completed = ochered["completed"];
        json obj;
        json *cheloveck = &obj;
        for (auto& ochered : jout) {
            if (ochered["userId"] == userID){
                cheloveck = &ochered;
                break;
            }
        }
        if (cheloveck -> empty()){
            jout.push_back({
            {"userId", userID},
            {"task_completed", 0}
        });
        cheloveck = &jout[jout.size()-1];
    }
    if (completed){
        (*cheloveck)["task_completed"] = static_cast<int>((*cheloveck)["task_completed"]) + 1;
    }
}

output << std::setw(2) << jout << endl;
}

File: Practice/24/Python/codejson.py
import json
with open('in.json', 'r') as f:
    j = json.load(f)
vivod = []

for zadacha in j:
    userID = zadacha['userId']
    complited = zadacha['completed']

    check = None
    for cheloveck in vivod:
        if cheloveck['userId'] == userID:
            check = cheloveck
            break
    
    if check == None:
        check = {}
        check['userId'] = userID
        check['task_completed'] = 0
        vivod.append(check)
    if complited:
        check['task_completed'] += 1
with open('out.json', 'w') as f:
    f.write(json.dumps(vivod, indent=2))